docs/csharp/programming-guide/unsafe-code-pointers/fixed-size-buffers.md

```diff
@@ -9,50 +9,48 @@ helpviewer_keywords:
-  
- For example, before C# 2.0, the following `struct` would be 8 bytes in size. The `pathName` array is a reference to the heap-allocated array:  
-  
- [!code-csharp[csProgGuidePointers#19](../../../csharp/programming-guide/unsafe-code-pointers/codesnippet/CSharp/fixed-size-buffers_1.cs)]  
-  
- Beginning with C# 2.0, a `struct` can contain an embedded array. In the following example, the `fixedBuffer` array has a fixed size. To access the elements of the array, you use a `fixed` statement to establish a pointer to the first element. The `fixed` statement pins an instance of `fixedBuffer` to a specific location in memory.  
-  
- [!code-csharp[csProgGuidePointers#20](../../../csharp/programming-guide/unsafe-code-pointers/codesnippet/CSharp/fixed-size-buffers_2.cs)]  
-  
- The size of the 128 element `char` array is 256 bytes. Fixed size [char](../../../csharp/language-reference/keywords/char.md) buffers always take two bytes per character, regardless of the encoding. This is true even when char buffers are marshaled to API methods or structs with `CharSet = CharSet.Auto` or `CharSet = CharSet.Ansi`. For more information, see <xref:System.Runtime.InteropServices.CharSet>.  
-  
- Another common fixed-size array is the [bool](../../../csharp/language-reference/keywords/bool.md) array. The elements in a `bool` array are always one byte in size. `bool` arrays are not appropriate for creating bit arrays or buffers.  
-  
+
+In C#, you can use the [fixed](../../language-reference/keywords/fixed-statement.md) statement to create a buffer with a fixed size array in a data structure. Fixed size buffers are useful when you write methods that interop with data sources from other languages or platforms. The fixed array can take any attributes or modifiers that are allowed for regular struct members. The only restriction is that the array type must be `bool`, `byte`, `char`, `short`, `int`, `long`, `sbyte`, `ushort`, `uint`, `ulong`, `float`, or `double`.
+
+```csharp
+private fixed char name[30];
+```
+
+## Remarks
+
+In safe code, a C# struct that contains an array does not contain the array elements. Instead, the struct contains a reference to the elements. You can embed an array of fixed size in a [struct](../../language-reference/keywords/struct.md) when it is used in an [unsafe](../../language-reference/keywords/unsafe.md) code block.
+
+The following `struct` is 8 bytes in size. The `pathName` array is a reference to the heap-allocated array:
+
+[!code-csharp[Struct with embedded array](../../../../samples/snippets/csharp/keywords/FixedKeywordExamples.cs#6)]
+
+A `struct` can contain an embedded array in unsafe code. In the following example, the `fixedBuffer` array has a fixed size. You use a `fixed` statement to establish a pointer to the first element. You access the elements of the array through this pointer. The `fixed` statement pins an instance of `fixedBuffer` to a specific location in memory.
```

Comment:
instance of fixedBuffer is imprecise; it's a character array of the MyBuffer instance, or the \`fixedBuffer\` instance field